### PR TITLE
feat: mailhog fixture

### PR DESCRIPTION
**Pull request by shivamklr**
## What does this PR do?
This PR is a followup on these PRs, #10521 and #10637.
This PR is a PoC for mailhog that can be used to assert emails.

## Type of change

- [x] Chore/Test (refactoring code, technical debt, workflow improvements)

## How should this be tested?
- Make sure the `mailhog` container is running. Ideally, it should run with`yarn dx`. 
- The default env variables should work `EMAIL_SERVER_HOST='localhost'` and `EMAIL_SERVER_PORT=1025`.
- Test `booking-pages.e2e.ts`

### CODE REVIEW

**Comment by nicktrn** [comment]
- #10535 
- #10560

**Review by leog — CHANGES_REQUESTED** [review]
Code looks good, very good addition to our test suite.
I just couldn't run it locally to verify it works as expected, getting a `Error: connect ECONNREFUSED ::1:8025` error. Is there any extra step needed to make it work?

**Comment by nicktrn** [comment]
@leog Mailhog needs to be running for this to work. I've taken a different approach in #10560 where playwright starts a mail server instance directly. Would be worth discussing pros and cons.

**Review by leog — APPROVED** [review]
Got it to work using `yarn dx` and since the changes to make it work in GH actions were taken in a different PR already merged, this looks good to me now. Thanks!

File: apps/web/playwright/fixtures/users.ts [hunk 1]
```diff
@@ -330,7 +331,19 @@ export const createUsersFixture = (page: Page, workerInfo: WorkerInfo) => {
...
+        const emailMessages = await emails.search(user.email);
+        if (emailMessages && emailMessages.count > 0) {
+          emailMessages.items.forEach((item) => {
+            emailMessageIds.push(item.ID);
+          });
+        }
+      }
+      for (const id of emailMessageIds) {
+        await emails.deleteMessage(id);
+      }
```
Comment (zomars) [inline apps/web/playwright/fixtures/users.ts:346]: @leog @shivamklr I think this is why I get so many errors running e2e locally on production builds.

image

Can we make this optional? Only check for mailhog if it's enabled. I spent quite a while figuring out why all my tests were failing locally. At least we should document and update the README with instructions.

Comment (shivamklr) [reply apps/web/playwright/fixtures/users.ts:346]: Sure. I will take a look and update the users fixture.

**Pull request merged**